### PR TITLE
Support the release entity for the GitHub provider

### PR DESCRIPTION
**Pull request by JAORMX**
# Summary

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [x] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [ ] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

File: internal/providers/github/properties/release.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+	tag, err := props.GetProperty(ReleasePropertyTag).AsString()
+	if err != nil {
+		return "", fmt.Errorf("failed to get tag name: %w", err)
+	}
+
+	return getReleaseNameFromParams(owner, repo, tag), nil
+}
+
+func getReleaseNameFromParams(owner, repo, tag string) string {
+	if owner == "" {
```
Comment (evankanderson) [inline internal/providers/github/properties/release.go:74]: This seems like a weird case -- does it actually happen, or could we put in a complaint?

**Review by evankanderson — COMMENTED** [review]
Does this create release entities, or just set us up for doing so in the future?

It looks (right now), like there's still more work before you can write a rule against these release objects -- are they created automatically (like PRs) after registering a repo?

I'm assuming (like PRs), that we will only start looking at releases that are modified after Minder is installed?

**Comment by JAORMX** [comment]
> Does this create release entities, or just set us up for doing so in the future?
> 

It creates the release entities.

> It looks (right now), like there's still more work before you can write a rule against these release objects -- are they created automatically (like PRs) after registering a repo?
> 
> I'm assuming (like PRs), that we will only start looking at releases that are modified after Minder is installed?


Like PRs, these are created after registering a repo, as they get created. You could already start writing policies for these; the main issue we have right now is that we don't have an API to get general entities like these.

File: internal/providers/github/properties/release.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+	"net/http"
+
+	go_github "github.com/google/go-github/v63/github"
+
+	"github.com/mindersec/minder/internal/entities/properties"
+	minderv1 "github.com/mindersec/minder/pkg/api/protobuf/go/minder/v1"
+	v1 "github.com/mindersec/minder/pkg/providers/v1"
+)
+
+// Release Properties
```
[the remark this answers is not in the document]
Comment (JAORMX) [reply internal/providers/github/properties/release.go:18]: that's right. This applies to all of our entity properties.

File: internal/providers/github/properties/release.go (hunk 1 above)
Comment (evankanderson) [reply internal/providers/github/properties/release.go:74]: Let's make this an error and/or log at warning/error -- I think it's valid input to the GitHub API, but we shouldn't be getting _output_ in this non-normalized form.

**Review by evankanderson — APPROVED** [review]

File: internal/providers/github/properties/release.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+	if fetchErr != nil {
+		if result != nil && result.StatusCode == http.StatusNotFound {
+			return nil, v1.ErrEntityNotFound
+		}
+		return nil, fmt.Errorf("failed to fetch release: %w", fetchErr)
+	}
+
+	return map[string]any{
+		properties.PropertyUpstreamID: properties.NumericalValueToUpstreamID(release.GetID()),
+		properties.PropertyName:       getReleaseNameFromParams(owner, repo, release.GetTagName()),
```
Comment (evankanderson) [inline internal/providers/github/properties/release.go:115]: It looks like GitHub's `name` here is non-unique, but may still be worth exposing for e.g. filtering with CEL.  (Apply / omit evalution only to releases with certain names)

File: internal/providers/github/properties/release.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+// Release Properties
+const (
+	// ReleasePropertyOwner represents the github owner
+	ReleasePropertyOwner = "github/owner"
+	// ReleasePropertyRepo represents the github repo
+	ReleasePropertyRepo = "github/repo"
+	// ReleasePropertyTag represents the github release tag name.
+	ReleasePropertyTag = "github/tag"
+	// ReleasePropertyBranch represents the github release branch
+	ReleasePropertyBranch = "github/branch"
```
Comment (evankanderson) [inline internal/providers/github/properties/release.go:27]: It was suggested we should have a common-top-level property for these, and for the commit SHA.

**Review by evankanderson — COMMENTED** [review]
A few more comments.

It looks like *GitHub* uses the tag name as the unique identifier for releases, and you can have duplicate "name" (titles, really) for releases, but that my not be universal -- it feels like `getReleaseNameFromParams` will probably need to vary based on provider.

**Review by evankanderson — APPROVED** [review]
One nit on https://github.com/mindersec/minder/pull/4921/files#r1871541959 that I'd like to see us track (because having two different name formats makes _everything else_ harder).

**Comment by evankanderson** [comment]
If we merge this, we should make sure to follow up with a PR to restore the -0.4% code coverage drop from adding this new code.

**Review by rdimitrov — APPROVED** [review]

**Pull request merged**